### DISCUSSION TITLE
How to know if instance?

### DISCUSSION THREAD

**Discussion by jmpangilinan** [vfsfitvnm/frida-il2cpp-bridge#365] [Q&A] [answered]
so I have a instance of an Object:

`let charInstance = dataManagerInstance.method<Il2Cpp.Object>('get_Character').invoke();`

```
AssemblyCSharp.class('Character').method('Update').implementation = function():void{
    //how would I know if this is the charInstance??
}
```

how would I know if the Update I intercepted is the instance of charInstance var?

I tried

`
this.handle == charInstance.handle
`

but I get no equal to the charInstance

EDIT:

nvm I'm being stupid again, I just need to compare with `this.handle.equals(charInstance.handle)`

**Comment by vfsfitvnm** [comment, marked as answer]
`Il2Cpp.` classes all inherits from the `NativeStruct` class, which has an `equals` method (it basically compares the handles like you do)
